controller/stats_reporter.go

```diff
@@ -171,17 +168,17 @@ func init() {
 		Description: "Depth of the work queue",
 		Measure:     workQueueDepthStat,
 		Aggregation: view.LastValue(),
-		TagKeys:     []tag.Key{reconcilerTagKey, NamespaceTagKey, ServiceTagKey, ConfigTagKey, RevisionTagKey},
+		TagKeys:     []tag.Key{reconcilerTagKey},
 	}, {
 		Description: "Number of reconcile operations",
 		Measure:     reconcileCountStat,
 		Aggregation: view.Count(),
-		TagKeys:     []tag.Key{reconcilerTagKey, successTagKey, NamespaceTagKey, ServiceTagKey, ConfigTagKey, RevisionTagKey},
+		TagKeys:     []tag.Key{reconcilerTagKey, successTagKey, NamespaceTagKey},
```

Comment:
If we don't define other keys (service, config, revision) here, what's your plan to insert them when reporting? IIUC, this PR is to support these metrics for stackdriver. If we use `knative_revision` as resource type, the metrics must have those resource labels.

Reply:
After some testing discovered stackdriver is adding this tags by default if not present

https://github.com/knative/pkg/blob/master/metrics/stackdriver_exporter.go#L263

Reply:
Absolutely, it should be up to the various importing engines which have specific schema needs to adapt the schema to their needs, rather than requiring that Knative contort the metrics export code to match the intersection of schema needs as more vendors integrate with different metrics backends.